src/billing/usage_billing_service.py

```
@@ -0,0 +1,433 @@
+    # 1. Initialize external service clients and persistence layer
+    polar_sdk = PolarSubscriptionManager()
+    db_client = ConceptualDatabaseClient() # Our conceptual database client
+
+    # 2. Initialize our custom usage billing service, injecting dependencies
+    billing_service = UsageBillingService(polar_sdk, db_client)
+
+    user_id_active = "user_123"
+    user_id_inactive = "user_456" # User with a cancelled subscription
+    user_id_new = "user_789" # A brand new user with no history
+
+    print(f"--- Scenario 1: Active User with Initial Credits and Usage ---")
```

Ruff (0.15.13)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)